hail/src/main/scala/is/hail/expr/ir/BlockMatrixIR.scala

```diff
@@ -58,7 +60,12 @@ object BlockMatrixIR {
 abstract sealed class BlockMatrixIR extends BaseIR {
   def typ: BlockMatrixType
 
-  def pyExecute(): BlockMatrix = ExecuteContext.scoped(execute)
+  def pyExecute(): BlockMatrix = {
+    ExecuteContext.scoped { ctx =>
+      val opt = Optimize.optimize(this, noisy = true, context = "BlockMatrixIR", ctx = Some(ctx))
```

Comment:
this should really go through a lowering pass -- there could totally be something like a TableAggregate in a value IR somewhere

Reply:
Ok, sounds good! I'll just replace this a call to LoweringPipeline.relationalLowerer?

Reply:
Yeah. And maybe call the execute through Interpret the way we do for MatrixIR/TableIR that still go through this legacy path?

Reply:
oh, good call. I looked for the analogous path but I couldn't find where it was called in python